meson.build

```diff
@@ -121,9 +120,18 @@ endif
 add_project_arguments(option_defines, language: 'cpp')
 subdir('lib')
 subdir('mist')#This "generates" the headers..
+subdir('lsp')
 subdir('generated')
 subdir('src')
 subdir('test')
-#doxygen = find_program('doxygen', required: false)
+doxygen = find_program('doxygen', required: false)
+if doxygen.found()
+  doxyfile = configure_file(output: 'Doxyfile', input: 'Doxyfile.in', configuration: {
+    'PACKAGE_VERSION': version,
+    'RELEASE' : release,
+  })
+  layout = custom_target('copy_layout', input: 'DoxygenLayout.xml', output: 'DoxygenLayout.xml', command: ['cp', '@INPUT@', '@OUTPUT@'])
+  run_target('docs', command: [doxygen, doxyfile], depends: layout)
```

Comment:
The only poor thing about this is that the generated Doxyfile will be in the build dir, which does have the headers in mist/ (and also some in generated/) but doesn't have access to the entire source :/